docs/testplan/latency_test.md

```diff
@@ -0,0 +1,99 @@
+
+2. **Establish BGP Sessions and Routes**
+
+   Follow the steps outlined in the multi-tier BGP test plan to:
+   - Set up the necessary IPv6 BGP sessions
+   - Advertise the desired IPv6 routes
+
+3. **Configure Traffic Flows**
+
+   Set up traffic items on the traffic generators to ensure:
+   - Every DUT port receives traffic
+   - Every route on the DUT is exercised
+
+   For example:
+   - For each of the four traffic generators, define two topologies, each containing four **physical** ports.
+   - Configure and enable Layer 2 and Layer 3 protocols for both topologies.
+   - In IXIA-1, define two uni-directional traffic items:
+     - One targeting IXIA-2
+     - One targeting IXIA-4
+     - Both set to full mesh mode
+     - Each at 60% line rate, with a packet size of 1024 bytes
+
+   - Similarly, in IXIA-2, define traffic items targeting IXIA-1 and IXIA-3.
+   - Repeat the pattern across all traffic generators to establish a robust traffic matrix.
+
+## Test Case 1: Full-mesh Latency Test
+
+1. **Baseline Latency Measurement**
+   - Start all the traffic items simultaneously and run them for 1 minute. Record latency statistics.
+   - Record latency statistics from each traffic item.
+   - Compare latency differences between the two traffic items sourced from each traffic generator.
+2. **Packet Size Variation**
+   - Repeat the baseline test using different frame sizes: 64 bytes, 1024 bytes, 4096 bytes, and 8192 bytes.
+   - Analyze how packet size affects latency.
+3. **Traffic Rate Scaling**
+   - Increase each traffic item’s rate from 60% to 70%, 80%, and 90% of the line rate.
+   - For each traffic rate, repeat the baseline and packet size variation tests.
+   - Observe how latency and packet loss change with increased traffic load.
+4. **RFC2889 Mode Evaluation**
+   - Enable the RFC2889 flag in the traffic generator settings.
+   - Repeat all previous tests under this mode.
+   - Measure whether this setting has any impact on latency or packet loss.
+
+## Test case 2: Latency Test with an Oversubscribed Port
+
+This test is conducted in a one-tier network to evaluate latency under an oversubscribed state.
+
+1. Assume the DUT has X ports connected to traffic generators. Assume the DUT has X ports connected to traffic generators. Select the last port on a traffic generator as the Rx port, and designate the remaining (X-1) ports as Tx ports.
+2. Define (X-1) traffic items, each assigned to a different Tx port, using a frame size of 86 bytes. Set each traffic stream's rate to `(line_rate × 110%) / X` to create an oversubscribed condition.
+3. Start all the traffic items simultaneously and run them for 1 minute. Record latency statistics.
+4. Repeat the test with frame size 1024 bytes, 4096 bytes, and 8192 bytes.
+5. Analyze the results by comparing latency measurements across different frame sizes, identifying any patterns or anomalies, and determining how oversubscription affects latency performance.
```

Comment:
analyze is not part of the test plan, the test plan is to simply record and report, so we can do post analysis.